<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.3.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"hanfc.top","root":"/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":true,"show_result":false,"style":"default"},"back2top":{"enable":true,"sidebar":false,"scrollpercent":true},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="PMAT is an efficient assembly toolkit for assembling plant mitogenomes using third-generation (HiFi&#x2F;CLR&#x2F;ONT) sequencing data. PMAT can also be used to assemble chloroplast genomes or animal mitogenome">
<meta property="og:type" content="article">
<meta property="og:title" content="PMAT">
<meta property="og:url" content="http://hanfc.top/2023/05/28/PMAT/index.html">
<meta property="og:site_name" content="云伴风行">
<meta property="og:description" content="PMAT is an efficient assembly toolkit for assembling plant mitogenomes using third-generation (HiFi&#x2F;CLR&#x2F;ONT) sequencing data. PMAT can also be used to assemble chloroplast genomes or animal mitogenome">
<meta property="og:locale" content="zh_CN">
<meta property="article:published_time" content="2023-05-28T12:21:02.000Z">
<meta property="article:modified_time" content="2023-11-15T06:22:15.893Z">
<meta property="article:author" content="Fuchuan Han">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="http://hanfc.top/2023/05/28/PMAT/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>
<link rel="stylesheet" href="//cdn.jsdelivr.net/gh/theme-next/theme-next-needmoreshare2@1/needsharebutton.min.css"><style>
#needsharebutton-postbottom {
  cursor: pointer;
  height: 26px;
  margin-top: 10px;
  position: relative;
}
#needsharebutton-postbottom .btn {
  border: 1px solid $btn-default-border-color;
  border-radius: 3px;
  display: initial;
  padding: 1px 4px;
}
</style>
  <title>PMAT | 云伴风行</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">云伴风行</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">与云作伴，伴风而行</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

  <a href="https://github.com/yunmika" class="github-corner" title="Follow me on GitHub" aria-label="Follow me on GitHub" rel="noopener" target="_blank"><svg width="80" height="80" viewBox="0 0 250 250" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="http://hanfc.top/2023/05/28/PMAT/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/image/icon.jpg">
      <meta itemprop="name" content="Fuchuan Han">
      <meta itemprop="description" content="学会珍惜，懂得感恩，干干净净，简简单单">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="云伴风行">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          PMAT
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>

              <time title="创建时间：2023-05-28 20:21:02" itemprop="dateCreated datePublished" datetime="2023-05-28T20:21:02+08:00">2023-05-28</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">更新于</span>
                <time title="修改时间：2023-11-15 14:22:15" itemprop="dateModified" datetime="2023-11-15T14:22:15+08:00">2023-11-15</time>
              </span>

          
            <div class="post-description">PMAT is an efficient assembly toolkit for assembling plant mitogenomes using third-generation (HiFi/CLR/ONT) sequencing data. PMAT can also be used to assemble chloroplast genomes or animal mitogenomes.</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="PMAT"><a href="#PMAT" class="headerlink" title="PMAT"></a><a name="h1">PMAT</a></h1><p>PMAT is an efficient assembly toolkit for assembling plant mitogenomes using third-generation (HiFi&#x2F;CLR&#x2F;ONT) sequencing data. PMAT can also be used to assemble chloroplast genomes or animal mitogenomes. </p>
<ul>
<li><a href="#h1">PMAT</a></li>
<li><a href="#C1">Installation</a></li>
<li><a href="#C2">Repuirement</a></li>
<li><a href="#C3">Options and usage</a><ul>
<li><a href="#C4">autoMito</a></li>
<li><a href="#C5">graphBuild</a></li>
</ul>
</li>
<li><a href="#C6">Examples</a><ul>
<li><a href="#C6.1">Demo1</a></li>
<li><a href="#C6.2">Demo2</a></li>
<li><a href="#C6.3">Demo3</a></li>
<li><a href="#C6.4">Demo4</a></li>
</ul>
</li>
<li><a href="#C7">Resulting files</a></li>
<li><a href="#C8">Version</a></li>
<li><a href="#C9">Citing PMAT</a></li>
</ul>
<h2 id="Installation"><a href="#Installation" class="headerlink" title="Installation "></a><a name="C1">Installation </a></h2><p>Install using git</p>
<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token function">git</span> clone https://github.com/bichangwei/PMAT.git
<span class="token builtin class-name">cd</span> PMAT/bin
<span class="token function">chmod</span> a+x PMAT
PMAT <span class="token parameter variable">--help</span><span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span></span></code></pre>
<p>Install by downloading the source codes</p>
<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token function">wget</span> https://github.com/bichangwei/PMAT/archive/refs/tags/v1.5.1.tar.gz
<span class="token function">tar</span> <span class="token parameter variable">-zxvf</span> v1.5.1.tar.gz
<span class="token builtin class-name">cd</span> PMAT-1.5.1/bin
<span class="token function">chmod</span> a+x PMAT
PMAT <span class="token parameter variable">--help</span><span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span></span></code></pre>

<h2 id="Requirement"><a href="#Requirement" class="headerlink" title="Requirement"></a><a name="C2">Requirement</a></h2><ul>
<li><a target="_blank" rel="noopener" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=Download"><strong>BLASTn</strong></a>  Needs to be installed in <code>PATH</code>.</li>
<li><a target="_blank" rel="noopener" href="https://github.com/YanshuQu/runAssembly"><strong>Singularity</strong></a> is required for PMAT versions earlier than 1.30. You can find installation instructions <a target="_blank" rel="noopener" href="https://github.com/YanshuQu/runAssembly">here</a>.</li>
<li><a target="_blank" rel="noopener" href="https://github.com/apptainer/apptainer/blob/main/INSTALL.md"><strong>Apptainer</strong></a> is required for PMAT versions 1.30 and later. Installation instructions can be found <a target="_blank" rel="noopener" href="https://github.com/apptainer/apptainer/blob/main/INSTALL.md">here</a>.</li>
<li><a target="_blank" rel="noopener" href="https://github.com/marbl/canu"><strong>Canu</strong></a> or <a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo"><strong>NextDenovo</strong></a> is required for CLR or ONT sequencing data, which is suggested to be installed in <code>PATH</code>.</li>
</ul>
<h2 id="Options-and-usage"><a href="#Options-and-usage" class="headerlink" title="Options and usage"></a><a name="C3">Options and usage</a></h2><p>Run <code>PMAT --help</code> to show the program’s usage guide.</p>
<pre class="line-numbers language-none"><code class="language-none">usage: PMAT &lt;command&gt; &lt;arguments&gt;

 ______     ___           __        ____       _____________ 
|   __  \  |   \        &#x2F;   |      &#x2F; __ \     |_____   _____|
|  |__)  | | |\ \      &#x2F; &#x2F;| |     &#x2F; &#x2F;  \ \          | |      
|   ____&#x2F;  | | \ \    &#x2F; &#x2F; | |    &#x2F; &#x2F;____\ \         | |      
|  |       | |  \ \  &#x2F; &#x2F;  | |   &#x2F; &#x2F;______\ \        | |      
|  |       | |   \ \&#x2F; &#x2F;   | |  &#x2F; &#x2F;        \ \       | |      
|__|       |_|    \__&#x2F;    |_| &#x2F;_&#x2F;          \_\      |_|      

PMAT            An efficient assembly toolkit for plant mitochondrial genome
Version         1.5.1
Contributors    Bi,C. and Han,F.
Email           bichwei@njfu.edu.cn, hanfc@caf.ac.cn

For more information about PMAT, please see https:&#x2F;&#x2F;github.com&#x2F;bichangwei&#x2F;PMAT

optional arguments:
-h, --help     show this help message and exit
-v, --version  show program&#39;s version and exit

Commands:

    autoMito    One-step de novo assembly of the mitochondrial genome. 
                This command can generate the master assembly graph 
                from raw sequencing data directly.

    graphBuild  If &#39;autoMito&#39; mode fails to generate the mitogenome 
                assembly graph, you can use this command to manually 
                select seeds for assembly.<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>

<h3 id="autoMito"><a href="#autoMito" class="headerlink" title="autoMito"></a><a name="C4">autoMito</a></h3><p>Run <code>PMAT autoMito --help</code> to view the usage guide.</p>
<pre class="line-numbers language-none"><code class="language-none">Required arguments:
  -i INPUT, --input INPUT
                        input raw sequencing file
  -o OUTPUT, --output OUTPUT
                        output directory
  -st SEQTYPE, --seqtype SEQTYPE
                        sequencing platform(ONT&#x2F;CLR&#x2F;HiFi)
  -g GENOMESIZE, --genomesize GENOMESIZE
                        Please enter the genome size of the species, such as 1G, 1000M.

optional arguments:
  -h, --help            show this help message and exit
  -tk TASK, --task TASK
                        all&#x2F;p1&#x2F; Default: all
                        all : De novo assembly including error correction for ONT&#x2F;CLR data and no error correction for HiFi data
                        p1  : Import error-corrected ONT&#x2F;CLR data for direct assembly
  -tp TYPE, --type TYPE
                        mt&#x2F;pt&#x2F;all Default: mt
                        mt   : Assembling the mitochondrial genome
                        pt   : Assembling the chloroplast genome
                        all  : Assembling the mitochondrial and chloroplast genomes
  -cs CORRECTSOFT, --correctsoft CORRECTSOFT
                        Correcting software using nextDenovo or Canu. Default: NextDenovo
  -cp CANU, --canu CANU
                        Please provide the install path of canu.
  -np NEXTDENOVO, --nextDenovo NEXTDENOVO
                        Please provide the install path of nextDenovo.
  -cfg CORRECTCFG, --correctcfg CORRECTCFG
                        config file for nextdenovo correct
  -fc FACTOR, --factor FACTOR
                        Subset extraction of error-corrected ONT, CLR or HiFi data. Sampling ratio factor in 0-1. Default: 1
  -sd SUBSEED, --subseed SUBSEED
                        Sampling set random number seeds, Default: 6
  -bn BREAKNUM, --breaknum BREAKNUM
                        break long reads (&gt;30k) with this. Default: 20000
  -ml MINOVERLAPLEN, --minoverlaplen MINOVERLAPLEN
                        set minimum overlap length. Default: 40
  -mi MINIDENTITY, --minidentity MINIDENTITY
                        set minimum overlap identification. Default: 90
  -cpu CPU              The number of threads. Default: 8
  -l MINLINK, --minLink MINLINK
                        Filter according to the minimum link depth provided by the user
  -m, --mem             Flag to keep sequence data in memory to speed up cup time
  -v, --version         show program&#39;s version and exit<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>

<p><strong>Notes</strong>:</p>
<ol>
<li>Make sure BLASTn was installed in PATH.</li>
<li><code>-tk</code>: There are two options for this parameter: ‘all’ or ‘p1’. For ONT or CLR raw data, ‘all’ is required to correct reads errors and trim the raw data. For error corrected data of ONT&#x2F;CLR, you can set ‘p1’ to skip the correct step. For HiFi data, this parameter can be ignored.</li>
<li><code>-cs</code>: For ONT or CLR raw data, users should provide the -cs parameter to select the error correction software, default: Nextdenovo.</li>
<li><code>-cp</code>: When using Canu for error correction, users need to use -cp parameter to specify the installation path of Canu. This parameter can be ignored when Canu is added to PATH.</li>
<li><code>-np</code>: When using NextDenovo to correct errors, users need to use the -np parameter to specify the installation path of NextDenovo. In addition, you need to use canu to trim the data after NextDenovo error correction, so the -cp parameter is needed to specify the installation path of Canu. When NextDenovo and Canu are added to PATH, this parameter can be ignored.</li>
<li><code>-cfg</code>: When using NextDenovo error correction, the user needs to specify a config file with -cfg. The contents of the configuration file are recommended to check <a target="_blank" rel="noopener" href="https://nextdenovo.readthedocs.io/en/latest/QSTART.html">NextDenovo</a>. Also, it is recommended to add the parameter -b for correction_options in config file.</li>
<li><code>-fc</code>: This parameter can be used to randomly select a subset of sequencing data for error correction and assembly. Default: all data.</li>
<li><code>-ml</code>: Parameters used for assembly, the default setting is 40. Recommended setting: 40~200.</li>
<li><code>-mi</code>: Parameters used for assembly, the default setting is 90. Recommended setting: 90~98.</li>
</ol>
<h3 id="graphBuild"><a href="#graphBuild" class="headerlink" title="graphBuild"></a><a name="C5">graphBuild</a></h3><p>If PMAT fails to generate the assembly graph in ‘autoMito’ mode, you can use this command to manually select seeds for assembly.</p>
<p>Run <code>PMAT graphBuild --help</code> to view the usage guide.</p>
<pre class="line-numbers language-none"><code class="language-none">Required arguments:
  -c CONTIGGRAPH, --ContigGraph CONTIGGRAPH
                        PMATContigGraph.txt: a file that can get all connections between contigs.
  -a ALLCONTIGS, --AllContigs ALLCONTIGS
                        PMATAllContigs.fna: a file that can get all the information of contigs.
  -o OUTPUT, --output OUTPUT
                        output directory
  -gs GENOMESIZE, --genomesize GENOMESIZE
                        Please enter the genome size of the species, such as 1G, 1000M.
  -rs READSIZE, --readsize READSIZE
                        The read size or file for assembly, such as 5G or assembly_seq.cut20K.fasta.

optional arguments:
  -h, --help            show this help message and exit
  -tp TYPE, --type TYPE
                        mt&#x2F;pt&#x2F;all Default: mt
                        mt   : Assembling the mitochondrial genome
                        pt   : Assembling the chloroplast genome
                        all  : Assembling the mitochondrial and chloroplast genomes
  -cpu CPU              The number of threads. Default: 8
  -s SEEDS [SEEDS ...], --seeds SEEDS [SEEDS ...]
                        ContigID for extending. Multiple contigIDs should be separated by space. For example: 1 312 356
  -l MINLINK, --minLink MINLINK
                        Filter according to the minimum link depth provided by the user
  -v, --version         show the program&#39;s version number and exit<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>
<p><strong>Notes</strong>:</p>
<ol>
<li>Make sure BLASTn was installed in PATH.</li>
<li><code>-c</code>: PMATContigGraph.txt generated by autoMito command.</li>
<li><code>-a</code>: PMATAllContigs.fna generated by autoMito command.</li>
<li><code>-gs</code>: The genome size of the species.</li>
<li><code>-rs</code>: The amount of data used by the assembly, or provide assembly_seq.cut20K.fasta generated by the graphBuild command.</li>
<li><code>-s</code>: Manually select the seeds for the extension, it is recommended to use more than 3 seeds. Use spaces to split between different seed IDs, e.g. 1,312,356.</li>
</ol>
<h2 id="Examples"><a href="#Examples" class="headerlink" title="Examples"></a><a name="C6">Examples</a></h2><p><strong><a name="C6.1">Demo1</a></strong></p>
<ol>
<li>Download a simulated Arabidopsis thaliana HiFi dataset:<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token function">wget</span> https://github.com/bichangwei/PMAT/releases/download/v1.1.0/Arabidopsis_thaliana_550Mb.fa.gz<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
<li>then run the autoMito command for one-click assembly:<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash">PMAT autoMito <span class="token parameter variable">-i</span> Arabidopsis_thaliana_550Mb.fa.gz <span class="token parameter variable">-o</span> ./test1 <span class="token parameter variable">-st</span> hifi <span class="token parameter variable">-g</span> 120m <span class="token parameter variable">-m</span><span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
<li>then use the graphBuild command to manually select seeds for assembly (used when the autoMito command fails to get a GFA file automatically):<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token comment"># Based on the PMATContigGraph.txt file, manually select 3 or more contigs that match the depth of mitochondrial genome sequencing</span>
PMAT graphBuild <span class="token parameter variable">-c</span> ./test1/assembly_result/PMATContigGraph.txt <span class="token parameter variable">-a</span> ./test1/assembly_result/PMATAllContigs.fna <span class="token parameter variable">-gs</span> 125m <span class="token parameter variable">-rs</span> ./test1/subsample/assembly_seq.cut20K.fasta <span class="token parameter variable">-o</span> ./test1_gfa <span class="token parameter variable">-s</span> <span class="token number">343</span> <span class="token number">345</span> <span class="token number">905</span> <span class="token number">513</span> <span class="token number">1344</span><span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre></li>
<li>PMAT runtime for different number of threads</li>
</ol>
<pre class="line-numbers language-none"><code class="language-none">8 CPUs: 13m25.342s; 16 CPUs: 9m29.853s; 32 CPUs: 8m42.429s; 64 CPUs: 7m57.279s<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>

<p><strong><a name="C6.3">Demo2</a></strong></p>
<ol>
<li>Download a simulated Malus domestica HiFi dataset:<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token function">wget</span> https://github.com/bichangwei/PMAT/releases/download/v1.1.0/Malus_domestica.540Mb.fasta.gz<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
<li>then run the autoMito command for one-click assembly:<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash">PMAT autoMito <span class="token parameter variable">-i</span> Malus_domestica.540Mb.fasta.gz <span class="token parameter variable">-o</span> ./test3 <span class="token parameter variable">-st</span> hifi <span class="token parameter variable">-g</span> 703m <span class="token parameter variable">-m</span><span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
<li>then use the graphBuild command to manually select seeds for assembly (used when the autoMito command fails to get gfa automatically):<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token comment"># Based on the PMATContigGraph.txt file, manually select 3 or more contigs that match the depth of mitochondrial genome sequencing</span>
PMAT graphBuild <span class="token parameter variable">-c</span> ./test3/assembly_result/PMATContigGraph.txt <span class="token parameter variable">-a</span> ./test3/assembly_result/PMATAllContigs.fna <span class="token parameter variable">-gs</span> 225m <span class="token parameter variable">-rs</span> ./test3/subsample/assembly_seq.cut20K.fasta <span class="token parameter variable">-o</span> ./test3_gfa <span class="token parameter variable">-s</span> <span class="token number">1</span> <span class="token number">2</span> <span class="token number">15391</span><span aria-hidden="true" class="line-numbers-rows"><span></span><span></span></span></code></pre></li>
<li>PMAT runtime for different number of threads</li>
</ol>
<pre class="line-numbers language-none"><code class="language-none">8 CPUs: 21m12.306s; 16 CPUs: 12m14.663s; 32 CPUs: 7m58.749s; 64 CPUs: 6m48.915s<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre>

<p><strong><a name="C6.3">Demo3</a></strong></p>
<ol>
<li>Download tested CLR data for Phaseolus vulgaris using IBM Aspera:<pre class="line-numbers language-none"><code class="language-none">ascp -v -QT -l 400m -P33001 -k1 -i ~&#x2F;.aspera&#x2F;connect&#x2F;etc&#x2F;asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:&#x2F;vol1&#x2F;fastq&#x2F;SRR291&#x2F;006&#x2F;SRR2912756&#x2F;SRR2912756_subreads.fastq.gz .<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
<li>then run the autoMito command for one-click assembly (CLR):<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash">PMAT autoMito <span class="token parameter variable">-i</span> SRR2912756_subreads.fastq.gz <span class="token parameter variable">-o</span> ./test_clr <span class="token parameter variable">-st</span> clr <span class="token parameter variable">-g</span> 540m <span class="token parameter variable">-cs</span> nextDenovo <span class="token parameter variable">-np</span> path/nextDenovo <span class="token parameter variable">-cp</span> path/canu <span class="token parameter variable">-cfg</span> nextdenovo.cfg <span class="token parameter variable">-m</span><span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
</ol>
<p><strong><a name="C6.4">Demo4</a></strong></p>
<ol>
<li>Download tested ONT data for Populus deltoides using IBM Aspera:<pre class="line-numbers language-none"><code class="language-none">ascp -v -QT -l 400m -P33001 -k1 -i ~&#x2F;.aspera&#x2F;connect&#x2F;etc&#x2F;asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:&#x2F;vol1&#x2F;fastq&#x2F;SRR122&#x2F;038&#x2F;SRR12202038&#x2F;SRR12202038_1.fastq.gz  .<span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
<li>then run the autoMito command for one-click assembly (ONT):<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash">PMAT autoMito <span class="token parameter variable">-i</span> SRR12202038_1.fastq.gz <span class="token parameter variable">-o</span> ./test_ont <span class="token parameter variable">-st</span> ont <span class="token parameter variable">-g</span> 430m <span class="token parameter variable">-cs</span> nextDenovo <span class="token parameter variable">-np</span> path/nextDenovo <span class="token parameter variable">-cp</span> path/canu <span class="token parameter variable">-cfg</span> nextdenovo.cfg <span class="token parameter variable">-m</span><span aria-hidden="true" class="line-numbers-rows"><span></span></span></code></pre></li>
</ol>
<h2 id="Resulting-files"><a href="#Resulting-files" class="headerlink" title="Resulting files"></a><a name="C7">Resulting files</a></h2><ul>
<li>The output files include:<ul>
<li><code>*/subsample/assembly_seq_subset.1.0.fasta</code>, The subsampled data for assembly</li>
<li><code>*/subsample/assembly_seq.cut20K.fasta</code>, The trimmed data for assembly</li>
<li><code>*/assembly_result/PMATAllContigs.fna</code>, The assembly result contains contig sequences</li>
<li><code>*/assembly_result/PMATContigGraph.txt</code>, The assembly result contains contig linking relationships</li>
<li><code>*/assembly_result/PMAT_mt_raw.gfa</code>, The initial assembly graph of mitogenome</li>
<li><code>*/assembly_result/PMAT_mt_master.gfa</code>, The optimized assembly graph of mitogenome</li>
<li><code>*/assembly_result/PMAT_pt_raw.gfa</code>, The initial assembly graph of the chloroplast genome</li>
<li><code>*/assembly_result/PMAT_pt_master.gfa</code>, The optimized assembly graph of the chloroplast genome</li>
</ul>
</li>
</ul>
<h2 id="Version"><a href="#Version" class="headerlink" title="Version"></a><a name="C8">Version</a></h2><p>PMAT version 1.5.0 (23&#x2F;11&#x2F;14)<br><br>Updates:</p>
<ul>
<li>In this version, PMAT adds automatic unloop function.</li>
</ul>
<p>PMAT version 1.4.0 (23&#x2F;11&#x2F;12)<br><br>Updates:</p>
<ul>
<li>In this version, we add a ‘-tp’ parameter for separate mitochondrial and chloroplast genome assembly.</li>
</ul>
<p>PMAT version 1.3.0 (23&#x2F;9&#x2F;25)<br><br>Updates:</p>
<ul>
<li>In this version, Apptainer is used instead of Singularity. Please ensure that the Apptainer is correctly installed.</li>
<li>In this version, you can perform multiple tasks simultaneously.</li>
</ul>
<h2 id="Citing-PMAT"><a href="#Citing-PMAT" class="headerlink" title="Citing PMAT"></a><a name="C9">Citing PMAT</a></h2><p>Bi C, Shen F, Han F, Qu Y, et al. PMAT: an efficient plant mitogenome assembly toolkit using ultra-low coverage HiFi sequencing data. Unpublished. <br><br>Bi C, Qu Y, Hou J, Wu K, Ye N, and Yin T. (2022). Deciphering the multi-chromosomal mitochondrial genome of Populus simonii. Front. Plant Sci. 13:914635.doi:10.3389&#x2F;fpls.2022.914635.</p>
<h2 id="Author"><a href="#Author" class="headerlink" title="Author"></a>Author</h2><p>Changwei Bi, <a href="mailto:&#98;&#105;&#x63;&#x68;&#x77;&#x65;&#105;&#x40;&#110;&#106;&#x66;&#117;&#46;&#x65;&#100;&#x75;&#46;&#x63;&#x6e;">&#98;&#105;&#x63;&#x68;&#x77;&#x65;&#105;&#x40;&#110;&#106;&#x66;&#117;&#46;&#x65;&#100;&#x75;&#46;&#x63;&#x6e;</a><br>Fuchuan Han, <a href="mailto:&#104;&#x61;&#110;&#x66;&#99;&#x40;&#99;&#97;&#x66;&#46;&#97;&#99;&#x2e;&#x63;&#x6e;">&#104;&#x61;&#110;&#x66;&#99;&#x40;&#99;&#97;&#x66;&#46;&#97;&#99;&#x2e;&#x63;&#x6e;</a></p>

    </div>

    
    
    <div class="post-widgets">
      <div id="needsharebutton-postbottom">
        <span class="btn">
          <i class="fa fa-share-alt" aria-hidden="true"></i>
        </span>
      </div>
    </div>
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>本文作者： </strong>Fuchuan Han
  </li>
  <li class="post-copyright-link">
    <strong>本文链接：</strong>
    <a href="http://hanfc.top/2023/05/28/PMAT/" title="PMAT">http://hanfc.top/2023/05/28/PMAT/</a>
  </li>
  <li class="post-copyright-license">
    <strong>版权声明： </strong>本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> 许可协议。转载请注明出处！
  </li>
</ul>
</div>


      <footer class="post-footer">

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2023/05/28/Lab-sever/" rel="prev" title="Lab-sever usage instructions">
      <i class="fa fa-chevron-left"></i> Lab-sever usage instructions
    </a></div>
      <div class="post-nav-item">
    <a href="/2023/06/09/PMAT-toolkit/" rel="next" title="PMAT-植物线粒体基因组组装工具包">
      PMAT-植物线粒体基因组组装工具包 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#PMAT"><span class="nav-number">1.</span> <span class="nav-text">PMAT</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Installation"><span class="nav-number">1.1.</span> <span class="nav-text">Installation </span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Requirement"><span class="nav-number">1.2.</span> <span class="nav-text">Requirement</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Options-and-usage"><span class="nav-number">1.3.</span> <span class="nav-text">Options and usage</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#autoMito"><span class="nav-number">1.3.1.</span> <span class="nav-text">autoMito</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#graphBuild"><span class="nav-number">1.3.2.</span> <span class="nav-text">graphBuild</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Examples"><span class="nav-number">1.4.</span> <span class="nav-text">Examples</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Resulting-files"><span class="nav-number">1.5.</span> <span class="nav-text">Resulting files</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Version"><span class="nav-number">1.6.</span> <span class="nav-text">Version</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Citing-PMAT"><span class="nav-number">1.7.</span> <span class="nav-text">Citing PMAT</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Author"><span class="nav-number">1.8.</span> <span class="nav-text">Author</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Fuchuan Han"
      src="/image/icon.jpg">
  <p class="site-author-name" itemprop="name">Fuchuan Han</p>
  <div class="site-description" itemprop="description">学会珍惜，懂得感恩，干干净净，简简单单</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">3</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yunmika" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yunmika" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Fuchuan Han</span>
</div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

  <script src="//cdn.jsdelivr.net/gh/theme-next/theme-next-needmoreshare2@1/needsharebutton.min.js"></script>
  <script>
      pbOptions = {};
        pbOptions.iconStyle = "box";
        pbOptions.boxForm = "horizontal";
        pbOptions.position = "bottomCenter";
        pbOptions.networks = "Wechat,QQZone";
      new needShareButton('#needsharebutton-postbottom', pbOptions);
  </script>
</body>
</html>
