<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.3.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"hanfc.top","root":"/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":true,"show_result":false,"style":"default"},"back2top":{"enable":true,"sidebar":false,"scrollpercent":true},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="PMAT is an efficient toolkit for mitochondrial genome assembly. It can correct three-generation sequencing data using [canu](https:&#x2F;&#x2F;github.com&#x2F;marbl&#x2F;canu) or [NextDenovo](https:&#x2F;&#x2F;github.com&#x2F;Nextomics">
<meta property="og:type" content="article">
<meta property="og:title" content="PMAT">
<meta property="og:url" content="http://hanfc.top/2023/05/28/PMAT/index.html">
<meta property="og:site_name" content="云伴风行">
<meta property="og:description" content="PMAT is an efficient toolkit for mitochondrial genome assembly. It can correct three-generation sequencing data using [canu](https:&#x2F;&#x2F;github.com&#x2F;marbl&#x2F;canu) or [NextDenovo](https:&#x2F;&#x2F;github.com&#x2F;Nextomics">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="http://hanfc.top/2023/05/28/PMAT/PMAT.png">
<meta property="article:published_time" content="2023-05-28T12:21:02.000Z">
<meta property="article:modified_time" content="2023-07-23T12:43:48.087Z">
<meta property="article:author" content="Fuchuan Han">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://hanfc.top/2023/05/28/PMAT/PMAT.png">

<link rel="canonical" href="http://hanfc.top/2023/05/28/PMAT/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>
<link rel="stylesheet" href="//cdn.jsdelivr.net/gh/theme-next/theme-next-needmoreshare2@1/needsharebutton.min.css"><style>
#needsharebutton-postbottom {
  cursor: pointer;
  height: 26px;
  margin-top: 10px;
  position: relative;
}
#needsharebutton-postbottom .btn {
  border: 1px solid $btn-default-border-color;
  border-radius: 3px;
  display: initial;
  padding: 1px 4px;
}
</style>
  <title>PMAT | 云伴风行</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="切换导航栏">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">云伴风行</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">与云作伴，伴风而行</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>首页</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>归档</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>

  <a href="https://github.com/yunmika" class="github-corner" title="Follow me on GitHub" aria-label="Follow me on GitHub" rel="noopener" target="_blank"><svg width="80" height="80" viewBox="0 0 250 250" aria-hidden="true"><path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path><path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="http://hanfc.top/2023/05/28/PMAT/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/image/icon.jpg">
      <meta itemprop="name" content="Fuchuan Han">
      <meta itemprop="description" content="学会珍惜，懂得感恩，干干净净，简简单单">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="云伴风行">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          PMAT
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">发表于</span>

              <time title="创建时间：2023-05-28 20:21:02" itemprop="dateCreated datePublished" datetime="2023-05-28T20:21:02+08:00">2023-05-28</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">更新于</span>
                <time title="修改时间：2023-07-23 20:43:48" itemprop="dateModified" datetime="2023-07-23T20:43:48+08:00">2023-07-23</time>
              </span>

          
            <div class="post-description">PMAT is an efficient toolkit for mitochondrial genome assembly. It can correct three-generation sequencing data using [canu](https://github.com/marbl/canu) or [NextDenovo](https://github.com/Nextomics/NextDenovo) and then use [Newbler](https://evomics.org/learning/assembly-and-alignment/newbler/) to assemble it, and it will find the mitochondrial genome sequence and generate the mitochondrial genome structure.</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h1 id="PMAT"><a href="#PMAT" class="headerlink" title="PMAT"></a><a name="h1">PMAT</a></h1><p>an efficient assembly toolkit for plant mitochondrial genome</p>
<p><img src="/2023/05/28/PMAT/PMAT.png" alt="img"></p>
<p>PMAT is an efficient toolkit for mitochondrial genome assembly. It can correct three-generation sequencing data using <a target="_blank" rel="noopener" href="https://github.com/marbl/canu">canu</a> or <a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo">NextDenovo</a> and then use <a target="_blank" rel="noopener" href="https://evomics.org/learning/assembly-and-alignment/newbler/">Newbler</a> to assemble it. Finally, PMAT can identify mitochondrial genome sequences and generate mitochondrial genome structures.</p>
<ul>
<li><a href="#h1">PMAT</a></li>
<li><a href="#C1">Installation</a></li>
<li><a href="#C2">Repuirement</a></li>
<li><a href="#C3">options and usage</a></li>
<li><a href="#C4">autoMito</a></li>
<li><a href="#C5">graphBuild</a></li>
<li><a href="#C6">Tips</a></li>
<li><a href="#C7">Output files</a></li>
<li><a href="#C6">Version</a></li>
</ul>
<h2 id="Installation"><a href="#Installation" class="headerlink" title="Installation"></a><a name="C1">Installation</a></h2><h4 id="recommended-installation-method"><a href="#recommended-installation-method" class="headerlink" title="recommended installation method"></a>recommended installation method</h4><pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token function">wget</span> https://github.com/bichangwei/PMAT/releases/download/v1.1.0/PMAT-1.1.0.tar.gz
<span class="token function">tar</span> <span class="token parameter variable">-zxvf</span> PMAT-1.1.0.tar.gz
<span class="token builtin class-name">cd</span> PMAT/bin
<span class="token function">chmod</span> a+x PMAT
PMAT <span class="token parameter variable">--help</span><span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span></span></code></pre>

<h2 id="Repuirement"><a href="#Repuirement" class="headerlink" title="Repuirement"></a><a name="C2">Repuirement</a></h2><h6 id="When-inputting-HiFi-reads-the-following-software-are-required"><a href="#When-inputting-HiFi-reads-the-following-software-are-required" class="headerlink" title="When inputting HiFi reads, the following software are required:"></a>When inputting HiFi reads, the following software are required:</h6><ul>
<li><a target="_blank" rel="noopener" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=Download"><strong>blastn</strong></a>  Needs to be installed in the <code>PATH</code>.</li>
<li><a target="_blank" rel="noopener" href="https://github.com/YanshuQu/runAssembly"><strong>singularity</strong></a></li>
</ul>
<h6 id="When-inputting-CLR-or-ONT-reads-the-following-software-are-required"><a href="#When-inputting-CLR-or-ONT-reads-the-following-software-are-required" class="headerlink" title="When inputting CLR or ONT reads, the following software are required:"></a>When inputting CLR or ONT reads, the following software are required:</h6><ul>
<li><a target="_blank" rel="noopener" href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=Download"><strong>blastn</strong></a>     Needs to be installed in the <code>PATH</code>.</li>
<li><a target="_blank" rel="noopener" href="https://github.com/marbl/canu"><strong>canu</strong></a> or <a target="_blank" rel="noopener" href="https://github.com/Nextomics/NextDenovo"><strong>NextDenovo</strong></a>    Suggested to be installed in the <code>PATH</code>.</li>
<li><a target="_blank" rel="noopener" href="https://github.com/YanshuQu/runAssembly"><strong>singularity</strong></a></li>
</ul>
<h2 id="options-and-usage"><a href="#options-and-usage" class="headerlink" title="options and usage"></a><a name="C3">options and usage</a></h2><p>Run <code>PMAT --help</code> to view the program’s usage guide.</p>
<pre class="line-numbers language-none"><code class="language-none">usage: PMAT &lt;command&gt; &lt;arguments&gt;

 ______     ___           __        ____       _____________ 
|   __  \  |   \        &#x2F;   |      &#x2F; __ \     |_____   _____|
|  |__)  | | |\ \      &#x2F; &#x2F;| |     &#x2F; &#x2F;  \ \          | |      
|   ____&#x2F;  | | \ \    &#x2F; &#x2F; | |    &#x2F; &#x2F;____\ \         | |      
|  |       | |  \ \  &#x2F; &#x2F;  | |   &#x2F; &#x2F;______\ \        | |      
|  |       | |   \ \&#x2F; &#x2F;   | |  &#x2F; &#x2F;        \ \       | |      
|__|       |_|    \__&#x2F;    |_| &#x2F;_&#x2F;          \_\      |_|      

PMAT            an efficient assembly toolkit for plant mitochondrial genome
Version         1.1.0
Contributors    Bi,C. and Han,F.
Email           bichwei@njfu.edu.cn, hanfc@caf.ac.cn

For more information about PMAT, see https:&#x2F;&#x2F;github.com&#x2F;bichangwei&#x2F;PMAT

optional arguments:
-h, --help     show this help message and exit
-v, --version  show program&#39;s version number and exit

Commands:

    autoMito    One-step de novo assembly of mitochondrial genome. 
                This command corrects the raw ONT&#x2F;CLR data or uses 
                the corrected data or HiFi for assembly directly. 
                Based on the assembly result, automatically select 
                seeds for extension and filter false positives to 
                obtain an assembly map of the mitochondrial genome.

    graphBuild  If PMAT fails to generate the mitochondrial genome 
                assembly map in one-step assembly, you can use this 
                command to manually select seeds for assembly.<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>

<h3 id="autoMito"><a href="#autoMito" class="headerlink" title="autoMito"></a><a name="C4">autoMito</a></h3><p>One-step de novo assembly of mitochondrial genome. This command corrects the raw ONT&#x2F;CLR data or uses the corrected data or HiFi for assembly directly. Based on the assembly result, automatically select seeds for extension and filter false positives to obtain an assembly map of the mitochondrial genome.</p>
<p>run <code>PMAT autoMito --help</code> view the usage guide for the all command.</p>
<pre class="line-numbers language-none"><code class="language-none">Required arguments:
  -i INPUT, --input INPUT
                        input raw sequencing file
  -o OUTPUT, --output OUTPUT
                        output directory
  -st SEQTYPE, --seqtype SEQTYPE
                        sequencing platform(ONT&#x2F;CLR&#x2F;HiFi)
  -g GENOMESIZE, --genomesize GENOMESIZE
                        Please enter the genome size of the species, such as 1G, 1000M.

optional arguments:
  -h, --help            show this help message and exit
  -tk TASK, --task TASK
                        all&#x2F;p1&#x2F; Default: all
                        all : De novo assembly including error correction for ONT&#x2F;CLR data and no error correction for HiFi data
                        p1  : Import error-corrected ONT&#x2F;CLR or HiFi data for direct assembly
  -cs CORRECTSOFT, --correctsoft CORRECTSOFT
                        Correcting software using nextDenovo or Canu. Default: nextDenovo
  -cp CANU, --canu CANU
                        Please provide the install path of canu.
  -np NEXTDENOVO, --nextDenovo NEXTDENOVO
                        Please provide the install path of nextDenovo.
  -cfg CORRECTCFG, --correctcfg CORRECTCFG
                        config file for nextdenovo correct
  -fc FACTOR, --factor FACTOR
                        Subset extraction of error-corrected ONT, CLR or HiFi data. Sampling ratio factor in 0-1. Default&#x3D;1
  -sd SUBSEED, --subseed SUBSEED
                        Sampling set random number seeds, Default&#x3D;6
  -bn BREAKNUM, --breaknum BREAKNUM
                        break long reads (&gt;30k) with this. Default: 20000
  -ml MINOVERLAPLEN, --minoverlaplen MINOVERLAPLEN
                        set minimum overlap length. Default:40
  -mi MINIDENTITY, --minidentity MINIDENTITY
                        set minimum overlap identification. Default:90
  -cpu CPU              The number of threads. Default: 8
  -l MINLINK, --minLink MINLINK
                        Filter according to the minimum link depth provided by the user
  -v, --version         show program&#39;s version number and exit<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>

<p><strong>notes</strong>:</p>
<ol>
<li>Make sure blastn is installed in the PATH.</li>
<li>-tk : There are two options for this parameter, all and p1. For ONT or CLR raw data, all is required, and it can be used to correct errors and trim the raw data. For error corrected data, you can set p1 to go directly to the assembly step. For HiFi data, this parameter can be ignored.</li>
<li>-cs : If use ONT or CLR raw data, users need to provide the -cs parameter to select the error correction software, default use canu.</li>
<li>-cp : When using canu for error correction, users need to use -cp parameter to specify the installation path of canu. This parameter can be ignored when canu is added to the PATH.</li>
<li>-np : When using NextDenovo to correct errors, users need to use the -np parameter to specify the installation path of NextDenovo. In addition, you need to use canu to prune the data after NextDenovo error correction, so the -cp parameter is needed to specify the installation path of canu. When NextDenovo and canu are added to the PATH, this parameter can be ignored.</li>
<li>-cfg : When using NextDenovo error correction, the user needs to specify a config file with -cfg. The cONTents of the configuration file are recommended to check <a target="_blank" rel="noopener" href="https://nextdenovo.readthedocs.io/en/latest/QSTART.html">NextDenovo</a>. Also, it is recommended to add the -b parameter for correction_options in the config file.</li>
<li>-fc : This parameter can randomly select a subset of sequencing data for error correction and assembly. The default is using all data.</li>
<li>-sd : The seed used when extracting a subset of sequencing data that can be used with -fc.</li>
<li>-bn : Assembly requires reads of length less than 30000, recommended to use the default parameters.</li>
<li>-ml : Parameters used for assembly, default setting is 40. Recommended setting range is 40-200.</li>
<li>-mi : Parameters used for assembly, default setting is 90. Recommended setting range is 90-95.</li>
<li>-ml : Parameters used for filtering, recommended default settings.</li>
</ol>
<h3 id="graphBuild"><a href="#graphBuild" class="headerlink" title="graphBuild"></a><a name="C5">graphBuild</a></h3><p>If PMAT fails to generate the mitochondrial genome assembly map in one-step assembly, you can use this command to manually select seeds for assembly.</p>
<p>run <code>PMAT graphBuild --help</code> view the usage guide for the graphBuild command.</p>
<pre class="line-numbers language-none"><code class="language-none">Required arguments:
  -c CONTIGGRAPH, --ContigGraph CONTIGGRAPH
                        PMATContigGraph.txt: a file that can get all connections bewteen contigs.
  -a ALLCONTIGS, --AllContigs ALLCONTIGS
                        PMATAllContigs.fna: a file that can get all information of contigs.
  -o OUTPUT, --output OUTPUT
                        output directory
  -gs GENOMESIZE, --genomesize GENOMESIZE
                        Please enter the genome size of the species, such as 1G, 1000M.
  -rs READSIZE, --readsize READSIZE
                        The read size or file for assembly, such as 5G or assembly_seq_subset.0.1.fasta.

optional arguments:
  -h, --help            show this help message and exit
  -cpu CPU              The number of threads. Default: 8
  -s SEEDS [SEEDS ...], --seeds SEEDS [SEEDS ...]
                        ContigID for extending. Multiple contigIDs should be separated by space. For example: 1 312 356
  -l MINLINK, --minLink MINLINK
                        Filter according to the minimum link depth provided by the user
  -v, --version         show program&#39;s version number and exit<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre>
<p><strong>notes</strong>:</p>
<ol>
<li>Make sure blastn is installed in the PATH.</li>
<li>-c : PMATContigGraph.txt generated by autoMito command.</li>
<li>-a : PMATAllContigs.fna generated by autoMito command.</li>
<li>-gs : The genome size of the species.</li>
<li>-rs : The amount of data used by the assembly, or provide assembly_seq_subset.0.1.fasta generated by the graphBuild command.</li>
<li>-s : Manually select the seeds for the extension, it is recommended to use more than 3 seeds. Use spaces to split between different seed ids, e.g. 1,312,356.</li>
<li>-l : Parameters used for filtering, recommended default settings.</li>
</ol>
<h2 id="Tips"><a href="#Tips" class="headerlink" title="Tips"></a><a name="C6">Tips</a></h2><p>Tested using data that is already publicly available.</p>
<table>
<thead>
<tr>
<th>Species</th>
<th>Data type</th>
<th>SRA number</th>
</tr>
</thead>
<tbody><tr>
<td>Pohlia nutans</td>
<td>HiFi</td>
<td><a target="_blank" rel="noopener" href="https://ngdc.cncb.ac.cn/gsa/browse/CRA006048/CRR383826">CRR383826</a></td>
</tr>
<tr>
<td>Taxus chinensis</td>
<td>HiFi</td>
<td><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR14756467">SRR14756467</a></td>
</tr>
<tr>
<td>Juncus effusus</td>
<td>HiFi</td>
<td><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/sra/?term=ERR8282830">ERR8282830</a></td>
</tr>
<tr>
<td>Arabidopsis thaliana</td>
<td>HiFi</td>
<td><a target="_blank" rel="noopener" href="https://ngdc.cncb.ac.cn/gsa/browse/CRA004538/CRR302668">CRR302668</a></td>
</tr>
<tr>
<td>Helianthus annuus</td>
<td>HiFi</td>
<td><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR14782853">SRR14782853</a></td>
</tr>
<tr>
<td>Salix wilsonii</td>
<td>HiFi</td>
<td><a target="_blank" rel="noopener" href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR21570388">SRR21570388</a></td>
</tr>
</tbody></table>
<table>
<thead>
<tr>
<th align="left"><sub>Dataset<sub></sub></sub></th>
<th align="right"><sub>Size<sub></sub></sub></th>
<th align="right"><sub>Cov.<sub></sub></sub></th>
<th align="left"><sub>Asm options<sub></sub></sub></th>
<th align="right"><sub>CPU time<sub></sub></sub></th>
<th align="right"><sub>Wall time<sub></sub></sub></th>
<th align="right"><sub>RAM<sub></sub></sub></th>
<th align="right"><sub> N50<sub></sub></sub></th>
</tr>
</thead>
<tbody><tr>
<td align="left"><sub>[Mouse (C57&#x2F;BL6J)][mouse-data]</sub></td>
<td align="right"><sub>2.6Gb</sub></td>
<td align="right"><sub>&amp;times;25</sub></td>
<td align="left"><sub>-t48 -l0</sub></td>
<td align="right"><sub>172.9h</sub></td>
<td align="right"><sub>4.8h</sub></td>
<td align="right"><sub>76G</sub></td>
<td align="right"><sub>21.1Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Maize (B73)][maize-data]</sub></td>
<td align="right"><sub>2.2Gb</sub></td>
<td align="right"><sub>&amp;times;22</sub></td>
<td align="left"><sub>-t48 -l0</sub></td>
<td align="right"><sub>203.2h</sub></td>
<td align="right"><sub>5.1h</sub></td>
<td align="right"><sub>68G</sub></td>
<td align="right"><sub>36.7Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Strawberry][strawberry-data]</sub></td>
<td align="right"><sub>0.8Gb</sub></td>
<td align="right"><sub>&amp;times;36</sub></td>
<td align="left"><sub>-t48 -D10</sub></td>
<td align="right"><sub>152.7h</sub></td>
<td align="right"><sub>3.7h</sub></td>
<td align="right"><sub>91G</sub></td>
<td align="right"><sub>17.8Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Frog][frog-data]</sub></td>
<td align="right"><sub>9.5Gb</sub></td>
<td align="right"><sub>&amp;times;29</sub></td>
<td align="left"><sub>-t48</sub></td>
<td align="right"><sub>2834.3h</sub></td>
<td align="right"><sub>69.0h</sub></td>
<td align="right"><sub>463G</sub></td>
<td align="right"><sub>9.3Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Redwood][redwood-data]</sub></td>
<td align="right"><sub>35.6Gb</sub></td>
<td align="right"><sub>&amp;times;28</sub></td>
<td align="left"><sub>-t80</sub></td>
<td align="right"><sub>3890.3h</sub></td>
<td align="right"><sub>65.5h</sub></td>
<td align="right"><sub>699G</sub></td>
<td align="right"><sub>5.4Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Human (CHM13)][CHM13-data]</sub></td>
<td align="right"><sub>3.1Gb</sub></td>
<td align="right"><sub>&amp;times;32</sub></td>
<td align="left"><sub>-t48 -l0</sub></td>
<td align="right"><sub>310.7h</sub></td>
<td align="right"><sub>8.2h</sub></td>
<td align="right"><sub>114G</sub></td>
<td align="right"><sub>88.9Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Human (HG00733)][HG00733-data]</sub></td>
<td align="right"><sub>3.1Gb</sub></td>
<td align="right"><sub>&amp;times;33</sub></td>
<td align="left"><sub>-t48</sub></td>
<td align="right"><sub>269.1h</sub></td>
<td align="right"><sub>6.9h</sub></td>
<td align="right"><sub>135G</sub></td>
<td align="right"><sub>69.9Mb</sub></td>
</tr>
<tr>
<td align="left"><sub>[Human (HG002)][NA24385-data]</sub></td>
<td align="right"><sub>3.1Gb</sub></td>
<td align="right"><sub>&amp;times;36</sub></td>
<td align="left"><sub>-t48</sub></td>
<td align="right"><sub>305.4h</sub></td>
<td align="right"><sub>7.7h</sub></td>
<td align="right"><sub>137G</sub></td>
<td align="right"><sub>98.7Mb</sub></td>
</tr>
</tbody></table>
<pre class="line-numbers language-bash" data-language="bash"><code class="language-bash"><span class="token comment"># Examples of de novo assembly of mitochondrial genomes using Arabidopsis thaliana's publicly available HiFi data.</span>
PMAT autoMito <span class="token parameter variable">-i</span> CRR302668.fastq <span class="token parameter variable">-o</span> ./test <span class="token parameter variable">-st</span> hifi <span class="token parameter variable">-g</span> 125m <span class="token parameter variable">-fc</span> <span class="token number">0.1</span>

<span class="token comment"># Manual selection of seeds to generate assembly graphs.</span>
PMAT graphBuild <span class="token parameter variable">-c</span> assembly_result/PMATContigGraph.txt-a ./test/assembly_result/PMATAllContigs.fna <span class="token parameter variable">-gs</span> 125m <span class="token parameter variable">-rs</span> ./test/subsample/assembly_seq_subset.0.1.fasta <span class="token parameter variable">-o</span> ./test_gfa<span aria-hidden="true" class="line-numbers-rows"><span></span><span></span><span></span><span></span><span></span></span></code></pre>
<h2 id="Output-files"><a href="#Output-files" class="headerlink" title="Output files"></a><a name="C7">Output files</a></h2><p>PMAT will produce a series of folders with the results:</p>
<ul>
<li>.&#x2F;test&#x2F;subsample&#x2F;assembly_seq_subset.0.1.fasta -&gt; Data for assembly</li>
<li>.&#x2F;test&#x2F;subsample&#x2F;assembly_seq.cut20K.fasta -&gt; reads trimmed data</li>
<li>.&#x2F;test&#x2F;assembly_result&#x2F;PMATAllContigs.fna -&gt; The assembly result contains contig sequences</li>
<li>.&#x2F;test&#x2F;assembly_result&#x2F;PMATContigGraph.txt -&gt; The assembly result contains contig linking relationships</li>
<li>.&#x2F;test&#x2F;assembly_result&#x2F;PMAT_raw.gfa -&gt; Initial results of mitochondrial genome assembly for visualization</li>
<li>.&#x2F;test&#x2F;assembly_result&#x2F;PMAT_master.gfa -&gt; Mitochondrial genome assembly optimization results for visualization</li>
</ul>
<h2 id="Version"><a href="#Version" class="headerlink" title="Version"></a><a name="C8">Version</a></h2><p>PMAT version 1.1.0 (2023.05.05)</p>
<h2 id="Author"><a href="#Author" class="headerlink" title="Author"></a>Author</h2><p>Changwei Bi <a href="mailto:&#98;&#x69;&#99;&#104;&#119;&#x65;&#105;&#x40;&#x6e;&#106;&#x66;&#117;&#x2e;&#x65;&#x64;&#117;&#46;&#x63;&#110;">&#98;&#x69;&#99;&#104;&#119;&#x65;&#105;&#x40;&#x6e;&#106;&#x66;&#117;&#x2e;&#x65;&#x64;&#117;&#46;&#x63;&#110;</a><br>Fuchuan Han <a href="mailto:&#x68;&#97;&#110;&#x66;&#x63;&#x40;&#99;&#97;&#102;&#46;&#x61;&#99;&#46;&#99;&#x6e;">&#x68;&#97;&#110;&#x66;&#x63;&#x40;&#99;&#97;&#102;&#46;&#x61;&#99;&#46;&#99;&#x6e;</a></p>

    </div>

    
    
    <div class="post-widgets">
      <div id="needsharebutton-postbottom">
        <span class="btn">
          <i class="fa fa-share-alt" aria-hidden="true"></i>
        </span>
      </div>
    </div>
        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>本文作者： </strong>Fuchuan Han
  </li>
  <li class="post-copyright-link">
    <strong>本文链接：</strong>
    <a href="http://hanfc.top/2023/05/28/PMAT/" title="PMAT">http://hanfc.top/2023/05/28/PMAT/</a>
  </li>
  <li class="post-copyright-license">
    <strong>版权声明： </strong>本博客所有文章除特别声明外，均采用 <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> 许可协议。转载请注明出处！
  </li>
</ul>
</div>


      <footer class="post-footer">

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2023/05/28/Lab-sever/" rel="prev" title="Lab-sever usage instructions">
      <i class="fa fa-chevron-left"></i> Lab-sever usage instructions
    </a></div>
      <div class="post-nav-item">
    <a href="/2023/06/09/PMAT-toolkit/" rel="next" title="PMAT-植物线粒体基因组组装工具包">
      PMAT-植物线粒体基因组组装工具包 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          文章目录
        </li>
        <li class="sidebar-nav-overview">
          站点概览
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#PMAT"><span class="nav-number">1.</span> <span class="nav-text">PMAT</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#Installation"><span class="nav-number">1.1.</span> <span class="nav-text">Installation</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#recommended-installation-method"><span class="nav-number">1.1.0.1.</span> <span class="nav-text">recommended installation method</span></a></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Repuirement"><span class="nav-number">1.2.</span> <span class="nav-text">Repuirement</span></a><ol class="nav-child"><li class="nav-item nav-level-6"><a class="nav-link" href="#When-inputting-HiFi-reads-the-following-software-are-required"><span class="nav-number">1.2.0.0.0.1.</span> <span class="nav-text">When inputting HiFi reads, the following software are required:</span></a></li><li class="nav-item nav-level-6"><a class="nav-link" href="#When-inputting-CLR-or-ONT-reads-the-following-software-are-required"><span class="nav-number">1.2.0.0.0.2.</span> <span class="nav-text">When inputting CLR or ONT reads, the following software are required:</span></a></li></ol></li></ol></li></ol></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#options-and-usage"><span class="nav-number">1.3.</span> <span class="nav-text">options and usage</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#autoMito"><span class="nav-number">1.3.1.</span> <span class="nav-text">autoMito</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#graphBuild"><span class="nav-number">1.3.2.</span> <span class="nav-text">graphBuild</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Tips"><span class="nav-number">1.4.</span> <span class="nav-text">Tips</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Output-files"><span class="nav-number">1.5.</span> <span class="nav-text">Output files</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Version"><span class="nav-number">1.6.</span> <span class="nav-text">Version</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#Author"><span class="nav-number">1.7.</span> <span class="nav-text">Author</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Fuchuan Han"
      src="/image/icon.jpg">
  <p class="site-author-name" itemprop="name">Fuchuan Han</p>
  <div class="site-description" itemprop="description">学会珍惜，懂得感恩，干干净净，简简单单</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">3</span>
          <span class="site-state-item-name">日志</span>
        </a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/yunmika" title="GitHub → https:&#x2F;&#x2F;github.com&#x2F;yunmika" rel="noopener" target="_blank"><i class="fab fa-github fa-fw"></i>GitHub</a>
      </span>
  </div>
  <div class="cc-license motion-element" itemprop="license">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" class="cc-opacity" rel="noopener" target="_blank"><img src="/images/cc-by-nc-sa.svg" alt="Creative Commons"></a>
  </div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Fuchuan Han</span>
</div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

  <script src="//cdn.jsdelivr.net/gh/theme-next/theme-next-needmoreshare2@1/needsharebutton.min.js"></script>
  <script>
      pbOptions = {};
        pbOptions.iconStyle = "box";
        pbOptions.boxForm = "horizontal";
        pbOptions.position = "bottomCenter";
        pbOptions.networks = "Wechat,QQZone";
      new needShareButton('#needsharebutton-postbottom', pbOptions);
  </script>
</body>
</html>
